packages/powersync_core/lib/src/sync_status.dart

```diff
@@ -202,3 +215,170 @@ class UploadQueueStats {
     }
   }
 }
+
+@internal
+final class InternalSyncDownloadProgress {
+  final Map<BucketPriority, int> downloaded;
+  final Map<BucketPriority, int> target;
+
+  final int _totalDownloaded;
+  final int _totalTarget;
+
+  InternalSyncDownloadProgress(this.downloaded, this.target)
+      : _totalDownloaded = target.values.sum,
+        _totalTarget = target.values.sum;
+
+  factory InternalSyncDownloadProgress.fromZero(Checkpoint target) {
+    final targetOps = target.checksums.groupFoldBy<BucketPriority, int>(
+      (cs) => BucketPriority(cs.priority),
+      (prev, cs) => (prev ?? 0) + (cs.count ?? 0),
```

Comment:
Note that the count here is the _total_ count for the bucket, not the incremental count between checkpoints. We'll need to add a new field to the API to include the count since the last checkpoint.

Reply:
We're now tracking operations on the client-side instead.